lighthouse-core/lib/dom-helpers.js

```diff
@@ -1,46 +0,0 @@
-/**
- * @license Copyright 2017 Google Inc. All Rights Reserved.
- * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
- * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
- */
-'use strict';
-
-/**
- * @fileoverview Helpers to be used with driver.evaluateAsync(). That is, these
- * methods are intended to be injected into and run in the context of the page.
- */
-
-/* global document */
-
-// TODO: move to page-functions.js
-
-/**
- * @param {string=} selector Optional simple CSS selector to filter nodes on.
- *     Combinators are not supported.
- * @return {Array<Element>}
- */
-/* istanbul ignore next */
-function getElementsInDocument(selector) {
-  /** @type {Array<Element>} */
-  const results = [];
-
-  /** @param {NodeListOf<Element>} nodes */
-  const _findAllElements = nodes => {
-    for (let i = 0, el; el = nodes[i]; ++i) {
-      if (!selector || el.matches(selector)) {
-        results.push(el);
-      }
-      // If the element has a shadow root, dig deeper.
-      if (el.shadowRoot) {
-        _findAllElements(el.shadowRoot.querySelectorAll('*'));
-      }
-    }
-  };
-  _findAllElements(document.querySelectorAll('*'));
-
-  return results;
-}
-
-module.exports = {
-  getElementsInDocumentFnString: getElementsInDocument.toString(),
```

Comment:
@paulirish I can't remember if you were an advocate for all the page functions to be exposed as strings or if it was the opposite (or if your opinion has changed over time)

Reply:
hmmm i dont think i cared, but i _like_ the idea of them being exported as strings.

we can do in a followup. 👩‍🚒

Reply:
followup sg :)